dev/AppLifecycle/AppInstance.cpp

```diff
@@ -0,0 +1,415 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <AppInstance.h>
+#include <Microsoft.Windows.AppLifecycle.AppInstance.g.cpp>
+
+#include "ActivationRegistrationManager.h"
+#include "LaunchActivatedEventArgs.h"
+#include "ProtocolActivatedEventArgs.h"
+#include "FileActivatedEventArgs.h"
+#include "Association.h"
+#include "ExtensionContract.h"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::ApplicationModel::Activation;
+
+namespace winrt::Microsoft::Windows::AppLifecycle::implementation
+{
+    INIT_ONCE AppInstance::s_initOnce{};
+    winrt::com_ptr<AppInstance> AppInstance::s_current;
+
+    std::tuple<std::wstring, std::wstring> ParseCommandLine(std::wstring commandLine)
+    {
+        auto argsStart = commandLine.rfind(c_argumentPrefix);
+        if (argsStart == std::wstring::npos)
+        {
+            return { L"", L"" };
+        }
+
+        argsStart += 4;
```

Comment:
comment here (why 4?)

Reply:
`argsStart += wcslen(c_argumentPrefix);`

Of course if c_argumentPrefix was a #define that could just be `argsStart += (ARRAYSIZE(c_argumentPrefix) - 1);`. He who liveth by the string sword, dieth by the string sword...